src/pretix/api/views/checkin.py

```diff
@@ -414,280 +755,143 @@ def redeem(self, *args, **kwargs):
-                    # upload the same scan over and over again. Since we can't update all devices quickly,
-                    # here's a dirty workaround to make it stop.
-                    try:
-                        brand = self.request.auth.software_brand
-                        ver = parse(self.request.auth.software_version)
-                        legacy_mode = (
-                            (brand == 'pretixSCANPROXY' and ver < parse('0.0.3')) or
-                            (brand == 'pretixSCAN Android' and ver < parse('1.11.2')) or
-                            (brand == 'pretixSCAN' and ver < parse('1.11.2'))
-                        )
-                        if legacy_mode:
-                            return Response({
-                                'status': 'error',
-                                'reason': Checkin.REASON_ALREADY_REDEEMED,
-                                'reason_explanation': None,
-                                'require_attention': False,
-                                '__warning': 'Compatibility hack active due to detected old pretixSCAN version',
-                            }, status=400)
-                    except:  # we don't care e.g. about invalid version numbers
-                        pass
+class CheckinRPCRedeemView(views.APIView):
+    def post(self, request, *args, **kwargs):
+        if isinstance(self.request.auth, (TeamAPIToken, Device)):
+            events = self.request.auth.get_events_with_permission(('can_change_orders', 'can_checkin_orders'))
+        elif self.request.user.is_authenticated:
+            events = self.request.user.get_events_with_permission(('can_change_orders', 'can_checkin_orders'), self.request).filter(
+                organizer=self.request.organizer
+            )
+        else:
+            raise ValueError("unknown authentication method")
+
+        s = CheckinRPCRedeemInputSerializer(data=request.data, context={'events': events})
+        s.is_valid(raise_exception=True)
+        return _redeem_process(
+            checkinlists=s.validated_data['lists'],
+            raw_barcode=s.validated_data['secret'],
+            answers_data=s.validated_data.get('answers'),
+            datetime=s.validated_data.get('datetime') or now(),
+            force=s.validated_data['force'],
+            checkin_type=s.validated_data['type'],
+            ignore_unpaid=s.validated_data['ignore_unpaid'],
+            nonce=s.validated_data.get('nonce'),
+            untrusted_input=True,
+            user=self.request.user,
+            auth=self.request.auth,
+            expand=self.request.query_params.getlist('expand'),
+            pdf_data=self.request.query_params.get('pdf_data', 'false') == 'true',
+            questions_supported=s.validated_data['questions_supported'],
+            canceled_supported=True,
+            request=self.request,  # this is not clean, but we need it in the serializers for URL generation
+            legacy_url_support=True,
```

Comment:
shouldn't this be `False`, because its used by the new checkin rpc url?

Reply:
Oops, thanks!